// sp-rest-proxy/dist/core/BasicRouter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var proxy_1 = require("../utils/proxy");
var logger_1 = require("../utils/logger");
var BasicRouter = (function () {
    function BasicRouter(ctx, settings) {
        this.ctx = ctx;
        this.settings = settings;
        this.util = new proxy_1.ProxyUtils(ctx, settings);
        this.logger = new logger_1.Logger(settings.logLevel);
    }
    BasicRouter.prototype.getHttpClient = function () {
        this.spr = this.util.getCachedRequest(this.spr);
        return this.spr;
    };
    BasicRouter.prototype.transmitResponse = function (res, response) {
        this.logger.verbose(response.statusCode, response.body);
        res.status(response.statusCode);
        res.contentType(response.headers['content-type'] || '');
        res.send(response.body);
    };
    BasicRouter.prototype.transmitError = function (res, err) {
        var statusCode = err.statusCode, message = err.message, error = err.error;
        this.logger.verbose('Error', { statusCode: statusCode, message: message, error: error });
        var response = err.response || {
            statusCode: statusCode || 400,
            headers: {},
            body: message || 'Unknown error'
        };
        res.status(response.statusCode);
        res.contentType(response.headers['content-type'] || '');
        res.send(response.body);
    };
    return BasicRouter;
}());
exports.BasicRouter = BasicRouter;
